### DISCUSSION TITLE
Problem with Locales again

### DISCUSSION THREAD

**Discussion by BrojJedan** [firefly-iii/firefly-iii#6806] [Firefly III]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [ ] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely me, not you](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

Despite setting all locales i still get this error message:
local.ERROR: Could not set locale. The following array doesnt work:  ["sl_SI.utf8","sl_SI.UTF-8"] 

I have all the necessary locales installed.

### Expected behaviour

Recognise locales.

### Debug information

locale -a:

C.UTF-8
en_GB.utf8
en_US.utf8
fr_FR.utf8
hr_HR.utf8
nl_NL.utf8
sl_SI.utf8

### Additional info

My environment is Linux Debian with PHP 8 and Apache.

**Comment by JC5** [comment]
Which version of Firefly III are you running?

**Comment by BrojJedan** [comment]
Oh, sorry - I forgot to add it.
Version  is 5.7.17.

**Comment by JC5** [comment]
Check out the /debug page, it should show the result of the setlocale attempt. This is the code that is used:

```
$moneyResult = setlocale(LC_MONETARY, $localeArray);
```

That's something you can try locally as well with the array of locales you just shared. Most locale packages contain everything but if your locales contain date info only, it may fail. There's not much I can do from my end otherwise.

**Reply by BrojJedan** [reply]
If I understand this correctly - there is LC_MONETARY option in my locales:
LANG=en_GB.UTF-8
LANGUAGE=en_GB:en
LC_CTYPE="en_GB.UTF-8"
LC_NUMERIC="en_GB.UTF-8"
LC_TIME="en_GB.UTF-8"
LC_COLLATE="en_GB.UTF-8"
LC_MONETARY="en_GB.UTF-8"
LC_MESSAGES="en_GB.UTF-8"
LC_PAPER="en_GB.UTF-8"
LC_NAME="en_GB.UTF-8"
LC_ADDRESS="en_GB.UTF-8"
LC_TELEPHONE="en_GB.UTF-8"
LC_MEASUREMENT="en_GB.UTF-8"
LC_IDENTIFICATION="en_GB.UTF-8"
LC_ALL=

Even if I set locale settings to en_GB, it still fails.. and it is a default locale with everything that is needed.
(Could not set locale. The following array doesnt work:  ["en_GB.utf8","en_GB.UTF-8"] )

**Reply by JC5** [reply]
Is the php process allowed to access the locales? Are you running something like SE linux perhaps? I'm sorry, these things are hard for me to debug 😅

**Comment by BrojJedan** [comment]
I guess it was my PHP installation - I used 8.0.
Switched to 8.1 and everything is working correctly.. 

Thank you for the support! The product looks awesome and I didn't even start it using fully!

**Reply by JC5** [reply]
Glad to hear it, enjoy! Thanks for the compliments!
